File: abc/abc137/a.cpp
// https://atcoder.jp/contests/abc137/submissions/26232541

#include <bits/stdc++.h>
using namespace std;
int main() {
  int A, B;
     cin >> A >> B;

    int ans = max({ A + B, A - B, A * B });

    cout << ans << endl;
}

File: abc/abc137/a.java
// https://atcoder.jp/contests/abc137/submissions/43290669

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int a = stdIn.nextInt();
    int b = stdIn.nextInt();

    int[] v = {a + b, a - b, a * b};
    int ans = v[0];
    for(int i = 1; i < 3; i++) {
      if(ans < v[i]) {
        ans = v[i];
      }
    }

    System.out.println(ans);
  }
}
